### PR TITLE
Implement Phase 1 of the enhanced release system

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes the following tickets:

DOC-2263
DOC-894
DOC-2650
DOC-2649
DOC-2651

### CODE REVIEW

**Review by lnhsingh — APPROVED** [review]

**Review by ianjevans — APPROVED** [review]

**Review by jseldess — APPROVED** [review]
LGTM, with a nit

**Comment by ianjevans** [comment]
I worked with Nick to verify the instructions in the updated wiki page and that Netlify builds fail on Jekyll errors.

File: _layouts/page.html [hunk 1]
```diff
@@ -36,19 +40,42 @@ <h1 class="post-title-main">{{ page.title }}</h1>
...
+    {% assign m = m_raw | date: "%s" %} {% comment %} Format m_raw in seconds. {% endcomment %}
+    {% assign a = a_raw | date: "%s" %} {% comment %} Format a_raw in seconds. {% endcomment %}
+
+   {% if page.version and a < today %} {% comment %} If the assistance support expiration date has passed, show the unsupported message. {% endcomment %}
       {% include_cached unsupported-version.md %}
    {% endif %}
 
-   {% if page.url contains 'v20.2' %}
-      {% include_cached unsupported-version-soon.md %}
+   {% if page.version and a >= today and m < today %} {% comment %} If the maintenance support expiration has passed but the version is still within the assistance support period, show this message nad pass the assistance support expiration date.. {% endcomment %}
```
[the remark this answers is not in the document]
Comment (nickvigilante) [reply _layouts/page.html]: Done.

File: releases/index.md [hunk 2]
```diff
@@ -32,40 +51,40 @@ After downloading your desired release, learn how to [Install CockroachDB](../{{
...
-                    <code>cockroachdb/cockroach{% if release.version contains "-" %}-unstable{% endif %}:{{ release.version }}</code>
+                    <code>cockroachdb/cockroach{% if r.version contains "-" %}-unstable{% endif %}:{{ r.version }}</code>
                 </section>
                 <section class="filter-content" data-scope="source">
-                {% if release.no_source %}
+                {% if r.no_source == "true" %}
                     N/A
                 {% else %}
-                    <a href="https://binaries.cockroachdb.com/cockroach-{{ release.version }}.src.tgz">Source</a>
+                    <a href="https://binaries.cockroachdb.com/cockroach-{{ r.version }}.src.tgz">Source</a>
```
[the remark this answers is not in the document]
Comment (nickvigilante) [reply releases/index.md]: I've updated https://github.com/cockroachdb/cockroach/pull/76740 to handle this.

**Review by jseldess — APPROVED** [review]
LGTM!

**Pull request merged**